src/functions/asyncCredential/asyncCredentialHandler.test.ts

```diff
@@ -958,41 +961,78 @@ describe("Async Credential", () => {
     });
   });
 
-  describe("Session recovery service", () => {
-    describe("Given service returns an error response", () => {
-      it("Returns 500 Server Error", async () => {
-        const jwtBuilder = new MockJWTBuilder();
-        const event = buildRequest({
-          headers: { Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}` },
-          body: JSON.stringify({
-            state: "mockState",
-            sub: "mockSub",
-            client_id: "mockClientId",
-            govuk_signin_journey_id: "mockGovukSigninJourneyId",
-          }),
-        });
-        dependencies.getRecoverSessionService = () =>
-          new MockFailingRecoverSessionService(
-            env.SESSION_TABLE_NAME,
-            env.SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME,
-          );
+  describe("Session Service", () => {
+    describe("Session recovery", () => {
+      describe("Given service returns an error response", () => {
+        it("Returns 500 Server Error", async () => {
+          const jwtBuilder = new MockJWTBuilder();
+          const event = buildRequest({
+            headers: { Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}` },
+            body: JSON.stringify({
+              state: "mockState",
+              sub: "mockSub",
+              client_id: "mockClientId",
+              govuk_signin_journey_id: "mockGovukSigninJourneyId",
+            }),
+          });
+          dependencies.getSessionService = () =>
+            new MockSessionServiceGetSessionBySubFailure(
+              env.SESSION_TABLE_NAME,
+              env.SESSION_TABLE_SUBJECT_IDENTIFIER_INDEX_NAME,
+            );
 
-        const result = await lambdaHandler(event, dependencies);
+          const result = await lambdaHandler(event, dependencies);
 
-        expect(result).toStrictEqual({
-          headers: { "Content-Type": "application/json" },
-          statusCode: 500,
-          body: JSON.stringify({
-            error: "server_error",
-            error_description: "Server Error",
-          }),
+          expect(result).toStrictEqual({
+            headers: { "Content-Type": "application/json" },
+            statusCode: 500,
+            body: JSON.stringify({
+              error: "server_error",
+              error_description: "Server Error",
+            }),
+          });
+        });
+      });
+
+      describe("Given service returns success response", () => {
+        describe("Given response value is the sessionId string", () => {
```

Comment:
Do we need this describe block? The description isn't very clear and I'm unsure what it achieves.

Reply:
Gone